library/alloc/src/collections/vec_deque/mod.rs

```diff
@@ -2182,16 +2182,14 @@ impl<T, A: Allocator> VecDeque<T, A> {
     // This may panic or abort
     #[inline(never)]
     fn grow(&mut self) {
-        if self.is_full() {
```

Comment:
> All callers already check that the buffer is full before calling `grow()`

That makes sense, but as part of this change I suggest making that expectation more visible.

Perhaps something like
```rust
    // Double the buffer size, when it's full.
    // This may panic or abort
    #[inline(never)]
    fn grow(&mut self) {
        debug_assert!(self.is_full());
```

Reply:
I'm happy to move the comment outside of the function but I'm not sure that it really is an expectation though. The same way one may want to reserve early in anticipation for a bunch of pushes in a hot loop, we could want to grow before being full, even if we don't do that yet.

It's not a hill I would die on though, in case you think it's important.

Reply:
I think it would be helpful to add a comment explaining the inline(never), which is basically that we expect callers to only call this in somewhat-cold paths, gated behind a conditional like being full. Obviously, we might also call this in other cases, but I would expect us to want to consider whether the inline(never) is then appropriate, so the comment makes sense to me.